Tests/WitsmlExplorer.Api.Tests/Workers/AnalyzeGapWorkerTests.cs

```diff
@@ -0,0 +1,289 @@
+
+        var logDataWithGaps = GetTestLogDataWithGaps(isDepthLog);
+        var logDataWithoutGaps = GetTestLogDataWithoutGaps(isDepthLog);
+        witsmlClient.InSequence(mockSequence)
+            .Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.IsAny<OptionsIn>()))
+            .Returns((WitsmlLogs logs, OptionsIn options) => isLogDataWithGaps
+                ? Task.FromResult(GetTestWitsmlLogs(logDataWithGaps,isDepthLog, isIncreasing))
+                : Task.FromResult(GetTestWitsmlLogs(logDataWithoutGaps, isDepthLog, isIncreasing)));
+        
+        witsmlClient.InSequence(mockSequence)
+            .Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.IsAny<OptionsIn>()))
+            .Returns((WitsmlLogs logs, OptionsIn options) => isLogDataWithGaps
+                ? Task.FromResult(GetTestWitsmlLogs(logDataWithGaps,isDepthLog, isIncreasing))
+                : Task.FromResult(GetTestWitsmlLogs(logDataWithoutGaps, isDepthLog, isIncreasing)));
+
+        witsmlClient.InSequence(mockSequence)
+            .Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.IsAny<OptionsIn>()))
+            .Returns(Task.FromResult(new WitsmlLogs(){Logs = new List<WitsmlLog>() }));
+    }
+    
+    private static WitsmlLogs GetTestWitsmlLogs(WitsmlLogData logData, bool isDepthLog, bool isIncreasing)
+    {
+        logData.Data = isIncreasing
+            ? logData.Data.OrderBy(x => x.Data).ToList()
+            : logData.Data.OrderByDescending(x => x.Data).ToList();
+        
+        (int startIndex, int endIndex) = isIncreasing ? (51, 54) : (54, 51);
+        (string startDateTimeIndex, string endDateTimeIndex) = isIncreasing ? ("2023-04-19T00:00:00Z", "2023-04-19T00:00:02Z") : ("2023-04-19T00:00:02Z", "2023-04-19T00:00:00Z");
+
+        return new WitsmlLogs
+        {
+            Logs = new WitsmlLog
+            {
+                UidWell = WellUid,
+                UidWellbore = WellboreUid,
+                Uid = LogUid,
+                StartIndex = new WitsmlIndex(new DepthIndex(startIndex, "m")),
+                EndIndex = new WitsmlIndex(new DepthIndex(endIndex, "m")),
+                StartDateTimeIndex = startDateTimeIndex,
+                EndDateTimeIndex = endDateTimeIndex,
+                IndexCurve = new WitsmlIndexCurve() { Value = "Depth" },
+                LogData = logData,
+                Direction = isIncreasing ? WitsmlLog.WITSML_DIRECTION_INCREASING : WitsmlLog.WITSML_DIRECTION_DECREASING,
+                IndexType = isDepthLog ? WitsmlLog.WITSML_INDEX_TYPE_MD : WitsmlLog.WITSML_INDEX_TYPE_DATE_TIME,
+                LogCurveInfo = new List<WitsmlLogCurveInfo>()
+                {
+                    new WitsmlLogCurveInfo()
+                    {
+                        Mnemonic = "Depth",
+                        MinIndex = new WitsmlIndex(new DepthIndex(startIndex, "m")),
```

Comment:
At least on Petrolink Test Server, the minIndex is actually the index with the lowest value (even for decreasing logs). Not sure if this is the case on other servers tho. That could maybe be the reason why it's not working on decreasing logs for me?

Reply:
It's still not working for decreasing logs for me at least. But the tests pass.

Reply:
If I understand you correctly, then the min and max index are always the same (minIndex is the index with the lowest value, and maxIndex is the one with the highest value) regardless of the log direction, and this is how it should be implemented, right?

Reply:
Yes, that is correct. The WITSML_STORE_API_Version_1.4.1.1 states:

> minIndex and maxIndex are the mathematical min/max index of each log curve.

Reply:
fixed